### PR TITLE
Project Review

### PR DESCRIPTION
**Pull request by RicardoEgas**
For this capstone project, we implemented the following:

- Set up initial tools such as webpack and Jest;
- Followed Gitflow;
- Used ES6 modules;
- Used callbacks and promises;
- Chose an external API and implement it in the projects to display items;
- Hit the Involvement API for the 3 different tasks (Likes, comments, reservations);
- Also used counters for them;
- Make JavaScript code asynchronous.;
- Performed coding tests using Jest
- Performed multiple code reviews for team members;
- Presentation video - https://drive.google.com/file/d/1SgZO4JLL35RbnyYv1uc9PRUnEbImVpzq/view?usp=sharing;

### CODE REVIEW

File: src/tests/commentCount.test.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+const getCommentCount = async (itemId) => {
+  const response = await fetch(
+    `${apiUrl}apps/HrIKPRrYjrxS00NlIVCD/comments?item_id=${itemId}`,
+  );
+  const data = await response.json();
+  return data.length;
+};
+
+export default getCommentCount;
```
Comment (Whoistolu) [inline src/tests/commentCount.test.js:11]: - [ ] [REQUIRED] Kindly ensure that you have a test for have tests for all edge cases. Kindly think at least about the scenario in which there are 0 items on the page (the counter function should return 0) as stated by the project requirements.

**Review by Whoistolu — CHANGES_REQUESTED** [review]
.

File: src/modules/postReserve.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import * as resInterface from './reserveInterface.js';
+
+// const projectId = 'HrIKPRrYjrxS00NlIVCD';
+const url = 'https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/HrIKPRrYjrxS00NlIVCD/reservations/';
```
Comment (Whoistolu) [inline src/modules/postReserve.js:3]: - [ ] [REQUIRED] When I try to make a reservation, I get this error
Screenshot 2023-06-10 at 22 42 40

- Hint to resolve
1. Disable any browser extensions that may interfere with the request.
2. Verify that the URL is correct and accessible. You can try opening it directly in your browser and see if it returns the expected response.
3. Test the request in a different browser or an incognito/private browsing window to see if the issue persists.
 - [ ] [REQUIRED] In order to ensure that the UI for the reservation looks like the provided design template, kindly add a close (X) button at the right top of the page so a user will be able to close the popup after making a reservation.
 
### Your Design
Screenshot 2023-06-10 at 22 46 31

### Template Design
Screenshot 2023-06-10 at 22 49 24

File: src/tests/commentCount.test.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const getCommentCount = async (itemId) => {
+  const response = await fetch(
+    `https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/HrIKPRrYjrxS00NlIVCD/comments?item_id=${itemId}`,
+  );
+  const data = await response.json();
+  return data.length;
+};
+
+export default getCommentCount;
```
Comment (Whoistolu) [inline src/tests/commentCount.test.js:9]: - [ ] [REQUIRED] I really do not see any test case here team, kindly ensure that the counter functions cover unit tests using Jest for all edge cases. DO consider the scenario in which there are 0 items on the page (the counter function should return 0) as stated by the project requirements.
- [ ] [REQUIRED] The counters functions should get covered with unit tests using the Jest framework.

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @RicardoEgas , @Bhekisile and @ometman 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._
- [ ] [REQUIRED] Great effort so far, however, I was unable to add my comment to the project. The popup refused to come up. I discovered that the js file handling the comments did not get merged to the `development` branch. Let's properly check the code base and ensure to merge the left-out files.

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
